components/engine_rocks/src/sst.rs

```diff
@@ -81,6 +81,8 @@ impl Iterable for RocksSstReader {
 // FIXME: See comment on RocksSstReader for why this contains Rc
 pub struct RocksSstIterator(DBIterator<Rc<SstFileReader>>);
 
+unsafe impl Send for RocksSstIterator {}
```

Comment:
Is it safe to send an `Rc`?

Reply:
I find it was used in a single thread for now.

Reply:
Okay, but how we guarantee it won't be used in multi-thread later? 😂

Reply:
Is it possible to define a method `into_sync` and returns send-able iterator？

Reply:
I don't find a simple way to address it since `Rc` is `!Send`, `Arc<RcoksSstIterator>` is still `!Send` https://doc.rust-lang.org/std/sync/struct.Arc.html#impl-Send
Maybe we can add a way for creating `DBIterator<Arc<SstFileReader>>` in rust-rocksdb later.
We can add a comment here.

Reply:
I did not find a simple way to address it, `Arc<RocksSstIterator>` is still `!Send` since `Rc` is `!Send` https://doc.rust-lang.org/std/sync/struct.Arc.html#impl-Send
The best solution is adding a method for creating `DBIterator<Arc<SstFileReader>>` in rust-rocksdb, we can add a TODO here and try it later.